agents/go-agents/src/main/go/core/rest/router.go

```diff
@@ -0,0 +1,84 @@
+//
+// Copyright (c) 2012-2017 Codenvy, S.A.
+// All rights reserved. This program and the accompanying materials
+// are made available under the terms of the Eclipse Public License v1.0
+// which accompanies this distribution, and is available at
+// http://www.eclipse.org/legal/epl-v10.html
+//
+// Contributors:
+//   Codenvy, S.A. - initial API and implementation
+//
+
+package rest
+
+import (
+	"log"
+	"net/http"
+
+	"github.com/julienschmidt/httprouter"
+)
+
+// Router provides http requests routing capabilities
+type Router interface {
+	http.Handler
+}
+
+type routerImpl struct {
```

Comment:
I think it makes more sense to create a separate go file and move `httprouter` based implementation there + add routes registration to the contract of the `Router` interface.